cpp/src/arrow/ipc/read_write_test.cc

```diff
@@ -2908,6 +2910,50 @@ INSTANTIATE_TEST_SUITE_P(PreBufferingTests, PreBufferingTest,
                            }
                          });
 
+Result<std::shared_ptr<RecordBatch>> MakeBatchWithDictionaries(const int length) {
+  auto schema_ = ::arrow::schema({::arrow::field("i32", int32()),
+                                  ::arrow::field("i32d", dictionary(int8(), int32()))});
+  std::shared_ptr<Array> a0, a1;
+  RETURN_NOT_OK(MakeRandomInt32Array(length, false, arrow::default_memory_pool(), &a0));
+  RETURN_NOT_OK(MakeRandomInt32Array(length, false, arrow::default_memory_pool(), &a1));
+  EXPECT_OK_AND_ASSIGN(auto encoded_datum, compute::DictionaryEncode(a1));
```

Comment:
I don't think we can depend on compute from IPC, right? We could just make a random array as the dictionary, then make a random array of indices, and manually build the dictionary array.

Reply:
Ah, true.  Technically I guess we can't even use filesystem.  I was thinking it was fair game for tests to use extra dependencies but that doesn't make sense as the tests still need to compile if the module isn't built at all.

Reply:
I've removed the dependencies